pages/recommend.js

```diff
@@ -0,0 +1,93 @@
+import Head from "next/head";
+import styles from "../styles/Recommend.module.css";
+import React from "react";
+import SearchBar from "../components/home/SearchBar";
+import BookCard from "../components/recommended/BookCard";
+import { useRouter, withRouter } from "next/router";
+
+export default withRouter(
+  class Home extends React.Component {
+    constructor(props) {
+      super(props);
+      this.state = {
+        basedOn: {},
+        recommended: [],
+      };
+    }
+
+    handleCheckBox = (e) => {
+      this.setState({ [e.target.name]: !this.state[e.target.name] });
+    };
+
+    getRecommended = () => {
+      const book_id = this.props.router.asPath.split("book_id=")[1];
```

Comment:
Don't like it... Didn't find any other quick workaround and wanted to get it done

Reply:
https://coderrocketfuel.com/article/retrieve-query-string-parameters-from-a-url-in-next-js